src/rluni/robot/robot.py

```diff
@@ -0,0 +1,428 @@
+from rluni.motors.motors import MN6007, MN2806
+from rluni.utils import get_validated_config_value as gvcv
+from rluni.utils import load_config_file
+
+from . import teledata as td
+
+DEG_TO_RAD = math.pi / 180
+REV_TO_RAD = 2 * math.pi
+
+# Create a logger
+logger = logging.getLogger(__name__)
+
+class EnabledMotors(Enum):
+    NONE = 0
+    ROLL = 1
+    YAW = 2
+    ROLL_PITCH = 3
+    ALL = 4
+    NUM_CONFIGS = 5
+
+motors = namedtuple("motors", ["roll", "pitch", "yaw"])
+torques = namedtuple("torques", ["roll", "pitch", "yaw"])
+class RobotSystem:
+    """
+    The RobotSystem class manages the core functionalities of a robot, including initializing sensors,
+    actuators, and handling the control loop for real-time operations.
+
+    Attributes:
+        LOOP_TIME (float): The fixed period for the control loop in seconds (e.g., 0.01 for 100Hz).
+        WRITE_DUTY (float): The fraction of the loop period before actuators are updated
+        MAX_TORQUE (float): The maximum torque to request from motors, used for testing and safety constraints.
+        xmotor (MN6007 or None): The motor controller instance, if motors are started.
+        itr (int): An iteration counter for the control loop.
+    """
+
+    def __init__(
+        self,
+        send_queue: Queue,
+        receive_queue: Queue,
+        start_motors=True,
+        controller_type: str = "test",
+        config_file=None,
+        motor_config=None,
+    ):
+        self.send_queue = send_queue
+        self.receive_queue = receive_queue
+
+        # Load and parse the configuration file
+        self._load_config(config_file)
+
+        # Initialize IMU and sensor fusion
+        self.imu = ICM20948(config_file=self.imu_config)
+        self.sensor_fusion = AHRSfusion(
+            sample_rate=int(1 / self.LOOP_TIME), config_file=self.imu_config
+        )
+
+        # # Initialize motor controller (if enabled)
+        # self.xmotor = MN6007() if start_motors else None
+
+
+        # set self.motor_config using arg string
+        if motor_config == "none":
+            self.motor_config = EnabledMotors.NONE
+            self.motors = self.motors(None, None, None)
+        elif motor_config == "roll":
+            self.motors = self.motors(MN6007(1, "roll"), None, None)
+            self.motor_config = EnabledMotors.ROLL
+        elif motor_config == "yaw":
+            self.motors = self.motors(None, None, MN2806(5, "yaw"))
+            self.motor_config = EnabledMotors.YAW
+        elif motor_config == "roll_pitch":
+            self.motors = self.motors(MN6007(1, "roll"), MN6007(4, "pitch"), None)
+            self.motor_config = EnabledMotors.ROLL_PITCH
+        elif motor_config == "all":
+            self.motors = self.motros(MN6007(1, "roll"), MN6007(4, "pitch"), MN2806(5, "yaw"))
+            self.motor_config = EnabledMotors.ALL
+        else: # catch-all 
+            self.motors = self.motors(None, None, None)
+            self.motor_config = EnabledMotors.NONE
```

Comment:
```suggestion


    def _initialize_motors(self, motor_config):
        """
        Initialize motors based on the given configuration string.

        Args:
            motor_config (str): The motor configuration, e.g., "none", "roll", "yaw", "roll_pitch", or "all".

        Returns:
            Tuple[namedtuple, EnabledMotors]: A tuple containing the motors namedtuple and EnabledMotors enum.
        """
        motor_mapping = {
            "none": (EnabledMotors.NONE, motors(None, None, None)),
            "roll": (EnabledMotors.ROLL, motors(MN6007(1, "roll"), None, None)),
            "yaw": (EnabledMotors.YAW, motors(None, None, MN2806(5, "yaw"))),
            "roll_pitch": (EnabledMotors.ROLL_PITCH, motors(MN6007(1, "roll"), MN6007(4, "pitch"), None)),
            "all": (EnabledMotors.ALL, motors(MN6007(1, "roll"), MN6007(4, "pitch"), MN2806(5, "yaw"))),
        }

        # Handle invalid configurations gracefully
        if motor_config not in motor_mapping:
            motor_config = "none"

        config_enum, motor_instances = motor_mapping[motor_config]
        return motor_instances, config_enum
```

Reply:
This takes up too much space in the init, defer it to another function tp enhance the clarity of the __init__ call and also encapsulate the functionality of setting the motors